src/java/edu/slu/tpen/servlet/CanvasServlet.java

```diff
@@ -63,7 +67,14 @@ protected void doGet(HttpServletRequest req, HttpServletResponse resp) throws Se
                     resp.setHeader("Access-Control-Allow-Headers", "*");
                     resp.setHeader("Access-Control-Expose-Headers", "*"); //Headers are restricted, unless you explicitly expose them.  Darn Browsers.
                     resp.setHeader("Cache-Control", "max-age=15, must-revalidate");
-                    resp.getWriter().write(export(buildPage(f)));
+
+                    if (req.getHeader("Accept") != null && req.getHeader("Accept").contains("iiif/v3")) {
+                        resp.setHeader("Content-Type", "application/ld+json;profile=\"http://iiif.io/api/presentation/3/context.json\"");
+                        resp.getWriter().write(export(buildPage(f, "v3")));
+                    }
+                    else {
+                        resp.getWriter().write(export(buildPage(f)));
+                    }
                     resp.setStatus(SC_OK);
                 } else {
                     getLogger(CanvasServlet.class.getName()).log(SEVERE, null, "No ID provided for canvas");
```

Comment:
A note, not required to this PR, but a good place to express a style practice worth considering...

Guard clauses and returns can reduce the complexity of this type of logic and make the `if...else` simpler. Consider line 64 a following:

```
if (folioID is not > 0) log error and return //eliminates else at :79
if ("accept" header is missing or not /v3) classic response and return // takes the alternative out of the else
// making it through those two gates means that the Accept Header is set to v3 so
buildPage(v3)

Reply:
I believe Thang is looking at that logic in his branch now, for what it's worth.